### DISCUSSION TITLE
Is creation of listings possible via Terraform

### DISCUSSION THREAD

**Discussion by Tamilselvan-Pattabiraman** [Snowflake-Labs/terraform-provider-snowflake#2124] [Q&A] [answered]
I assume currently we don't have option for the  creation of listings via terraform. As Listing SQL API is in private preview, we could build the SQL statements and if terraform as resource to execute SQL statements then we can achieve it. Is it possible to execute the SQL statements via this provider? We basically trying to avoid UI for the creation of listing so trying all other options.

**Comment by kolyarice** [comment]
now that the listings SQL API is in public preview, any sense of when it will be supported by the provider?
https://other-docs.snowflake.com/progaccess/listing-progaccess-about

**Reply by sfc-gh-asawicki** [reply]
We will share a detailed roadmap for the next months next week; listings and all other public preview features for that matter will be implemented after we correctly support the generally available features. But we will welcome contributions (we shall release contribution guidelines next week to ease the process).

**Reply by dluo-sig** [reply]
Is there any update on this feature request?  Would love to have everything managed in one place.

**Reply by sfc-gh-jcieslak** [reply, marked as answer]
Hey @dluo-sig 👋 the support for listings was added in the latest provider version (v2.5.0). Here's the documentation for it: https://registry.terraform.io/providers/snowflakedb/snowflake/latest/docs/resources/listing.

**Reply by dluo-sig** [reply]
> Hey @dluo-sig 👋 the support for listings was added in the latest provider version (v2.5.0). Here's the documentation for it: https://registry.terraform.io/providers/snowflakedb/snowflake/latest/docs/resources/listing.

Oh perfect, I must have just missed it.
